front/src/modules/pathfinding/components/Itinerary/DisplayItinerary/Vias.tsx

```diff
@@ -19,6 +28,74 @@ const Vias = ({ zoomToFeaturePoint }: ViasProps) => {
   const vias = useSelector(getVias());
   const pathSteps = useSelector(getPathSteps);
   const { launchPathfinding } = useManageTrainScheduleContext();
+  const infraId = useInfraID();
+  const [postSearch] = osrdEditoastApi.endpoints.postSearch.useMutation();
+  const dispatch = useAppDispatch();
+
+  const hasUpdatedNames = useRef(false);
+
+  const searchOperationalPointsByUIC = useCallback(async () => {
+    if (!infraId || !pathSteps.length) return [];
+
+    const validPathSteps = pathSteps.filter(
+      (step): step is PathStep & { uic: number; secondary_code?: string } =>
+        step !== null && 'uic' in step
+    );
+
+    if (!validPathSteps.length) return [];
+
+    const uicChConditions = validPathSteps.map(({ uic, secondary_code }) => [
+      'and',
+      ['=', ['uic'], uic],
```

Comment:
This handles the UIC case, but not other cases. Other cases might be used, e.g. trigram.

`buildOpSearchQuery()` already handles all of the cases, maybe we could just use that?